addons/crm/models/calendar.py

```diff
@@ -24,3 +24,12 @@ def create(self, vals):
         if event.opportunity_id:
             event.opportunity_id.log_meeting(event.name, event.start, event.duration)
         return event
+
+    # Due to same model crm.lead need to give opportunity view when opening document from event.
+    def action_open_resource_document(self):
```

Comment:
As the event has a link to opportunity_id, shouldn't we use that link instead in the override ?